normalize_license_text/normalize_class.py

```diff
@@ -0,0 +1,105 @@
+import re
+import sys
+
+""" This is the main class responsible for Normalization of License Texts. It takes into 
+account the case of the text, punctuations, copyright symbols and whitespaces. The Class mainly
+takes as input argument a string and normalizes into an output normalized_string. """ 
+
+class NormalizeText:
+    
+    def __init__(self,inputstring):
+        self.inputstring = inputstring
+        
+    def returnfinalstring(self):
+        normalized_string = ''
+        normalized_string = self.lowercase()
+        normalized_string = self.equivalentwords(normalized_string)
+        normalized_string = self.copyrightsymbol(normalized_string)
+        normalized_string = self.bullets_numbering(normalized_string)
+        normalized_string = self.punctuation(normalized_string)
+        normalized_string = self.license_title(normalized_string)
+        normalized_string = self.remove_whitespace(normalized_string)
+        return normalized_string
+        
+    def lowercase(self):
+        normalized_string = self.inputstring.lower()
+        return normalized_string
+        
+    def equivalentwords(self,normalized_string):
+        try:
+            equivalentfile = open("equivalentwords.txt","r")
+            for word in equivalentfile.readlines():
+                splitwords = word.split(',')
+                wordtoreplace = splitwords[0]
+                wordreplaced = splitwords[1]
+                wordtoreplace = str(wordtoreplace)
+                wordreplaced = str(wordreplaced)[:-1]
+                normalized_string = re.sub(wordreplaced,wordtoreplace,normalized_string)
+            equivalentfile.close()
+            return normalized_string
+        except IOError:
+            print("There was some problem with the function.")
+        
+        
+    def copyrightsymbol(self,normalized_string):
+        
+        """ Removes the copyright symbol and the possibilities of mismatch due to it. """
+        try:
+            normalized_string = normalized_string.replace("copyright",'')
+            normalized_string = normalized_string.replace('(c)','')
+            return normalized_string
+        except IOError:
+            print("This function could not run properly.")
+                    
+    def bullets_numbering(self,normalized_string):
+        
+        """ The most important and error prone function is the handling of the bullets
+        and numbering cases. The case has just 1 error prone possibility and requirement
+        that the numbering should carry a space after it or else it gets matched with the 
+        version. 
+        (1. ) will match while (1.) will not.
+        """
+        try:
+            normalized_string = re.sub(r'([0-9]+\.){2,}','',normalized_string)
```

Comment:
Keep all the regex in a list and call a for loop on that list, will trim down a lot of duplicate code

Reply:
@rtgdk I have optimized the code to a large extent and all the strings to be replaced are in a list. But the same can't be done with regex since when I store all the regexes in a list, the re complier reads it as a string and not a regex which results in a mismatch.

Reply:
You don't have to store the regex in the list, strings would work. 1 param for re.sub can be a string, it will internally convert it to "regex"

Reply:
@rtgdk  The problem I am facing is it the regex starts with a r' keyword to avoid recursive call and multiple '\'. When I add the r keyword in the list it identifies it as a string but when I remove it I would have to add multiple '\' which would make it even more complex. 
For example. ["([0-9]+\\\\)"] This is the regex to match 2\ if I put in a list. I could avoid the 4 times '\' in regex by using the r' keyword if I do it without using a list.

Reply:
@anshuldutt21 I didn't get the example you said. I tried out the following code and it worked fine on test string:
> l = ['([0-9]+\.){2,}', '[0-9]+\.[\D]',  '[a-z]\)',  '[0-9]\)']
normalized_string = 'test2.4.3.test2.atestb)5)test'
for i in l:
     normalized_string = re.sub(i, '', normalized_string)

returned the same output as

> normalized_string = re.sub(r'([0-9]+\.){2,}','',normalized_string)
normalized_string = re.sub(r'[0-9]+\.[\D]','',normalized_string)
normalized_string = re.sub(r'[a-z]\)','',normalized_string)
normalized_string = re.sub(r'[0-9]\)','',normalized_string)

Reply:
Done the changes. I will merge this PR now to track future progress.